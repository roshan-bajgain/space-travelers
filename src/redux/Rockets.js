import {
    GET_ROCKETS,
    SET_ROCKETS,
    SET_RESERVES,
    CANCEL_RESERVES,
  } from "../types";
  
  const initialState = { rockets: null };
  export default function rocketsReducer(state = initialState, action) {
    switch (action.type) {
      case GET_ROCKETS:
        return state;
      case SET_ROCKETS:
        return { ...state, rockets: action.payload };
      case SET_RESERVES:
        const newState = state.rockets.map((rocket) => {
          if (rocket.id !== action.payload) {
            return rocket;
          }
          return { ...rocket, reserved: true };
        });
        return { ...state, rockets: newState };
      case CANCEL_RESERVES:
        const newRocketsState = state.rockets.map((rocket) => {
          if (rocket.id !== action.payload) {
            return rocket;
          }
          return { ...rocket, reserved: false };
        });
        return { ...state, rockets: newRocketsState };
  
      default:
        return state;
    }
  }