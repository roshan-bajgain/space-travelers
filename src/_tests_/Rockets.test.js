import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../redux/configureStore";
import Rockets from "../components/Space-rockets";

describe("Rocket Element Tests", () => {
  test("Check if Component Renders in DOM", () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>
    );
    expect(screen.queryByText(/Missions/)).toBeNull();
    expect(screen.queryByText(/Description/)).toBeNull();
  });
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
