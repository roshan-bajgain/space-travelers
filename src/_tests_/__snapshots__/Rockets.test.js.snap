// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket Element Tests renders correctly 1`] = `
<section
  className="inner-padding"
/>
`;
